### PR TITLE
Terra install docs moved to main docs

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary
Terra install docs are now in the main doc repo.  The docs have also been updated for the Cython build process.

### CODE REVIEW

**Comment by jaygambetta** [comment]
are we in terra for installing by the source are we going to link back to this?

**Comment by nonhermitian** [comment]
I thought the idea discussed yesterday was to have a link from Terra to the main docs for the install process a la TensorFlow.

**Comment by jaygambetta** [comment]
yeah but then we need to fix the contributing.rst in terra. I dont think it is good to have two versions of the same thing. 

I am happy with this proposal but coordinate it in terra.

**Comment by nonhermitian** [comment]
Sure, that is the plan. First we need Matthew to merge the cython swapper. Then these get merged. Then i make pr to modify the terra contrib file.

**Comment by nonhermitian** [comment]
The Cython mapper has been merged in Terra.  Now these instructions are valid, and can be merged.

**Review by derivation — APPROVED** [review]

File: docs/install.rst [hunk 1]
```diff
@@ -73,7 +73,7 @@ Install from Source
 To get involved with the development of Qiskit, you can install any of the packages from source.
 See the contributing guide for each element of Qiskit for complete instructions:
 
-*	`Qiskit Terra <https://github.com/Qiskit/qiskit-terra/blob/master/.github/CONTRIBUTING.rst>`_
+*	`Qiskit Terra <install/terra.html>`_
```
Comment (arthur-1) [inline docs/install.rst:76]: `install/terra.html` is currently a dead link -- is this intentional?

Comment (jaygambetta) [reply docs/install.rst:76]: @nonhermitian can you fix this.

**Pull request merged**